File: 1666-make-the-string-great/make-the-string-great.cs
using System.Text;

public class Solution {
    public string MakeGood(string s) {
        StringBuilder sb = new StringBuilder(s);
        int flag = 0;
        while (flag == 0 && sb.Length > 0) {
            flag = 1;
            for (int i = 0; i < sb.Length - 1; i++) {
                if (IsGreat(sb, i)) {
                    sb.Remove(i, 2);
                    flag = 0;
                }
            }
        }
        return sb.ToString();
    }

    public bool IsGreat(StringBuilder sb, int i) {
        return (int)sb[i] == (int)sb[i + 1] + 32 || (int)sb[i] == (int)sb[i + 1] - 32;
    }
}

File: 1666-make-the-string-great/make-the-string-great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {
    let sb = s;
    let flag = 0;
    while (flag === 0 && sb.length > 0) {
        flag = 1;
        for (let i = 0; i < sb.length - 1; i++) {
            if (isGreat(sb, i)) { 
                sb = sb.substring(0, i) + sb.substring(i + 2);
                flag = 0;
                break; 
            }
        }
    }
    return sb;
}

function isGreat(sb, i) { // Changed the function declaration to standalone

    return Math.abs(sb.charCodeAt(i) - sb.charCodeAt(i + 1)) === 32;
}